### PR TITLE
DOCSP-37087 Kafka Concept page and TOC

### PR DESCRIPTION
**Pull request by jocelyn-mendez1**
## DESCRIPTION
Creates the Kafka concept page and layout kafka related toc with empty pages.

## STAGING
https://preview-mongodbjocelynmendez1.gatsbyjs.io/docs-relational-migrator/DOCSP-37087/installation/kafka-deployments/migrator-with-kafka/

## JIRA
https://jira.mongodb.org/browse/DOCSP-37087?filter=-1

## BUILD LOG
https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=65f474566fc5f04850dbc8e4

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/installation/kafka-deployments/migrator-with-kafka.txt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+   :class: singlecol
+
+You can use Relational Migrator to migrate data to and from Kafka. 
+Relational Migrator can be used as a Kafka Connect plug-in that improves 
+resilience and scalability of big data migration jobs.
+
+Use Cases
+---------
+
+Relational Migrator with Kafka is intended for large and long running jobs.
```
Comment (sarah-olson-mongodb) [inline source/installation/kafka-deployments/migrator-with-kafka.txt:22]: (content updates)
Please be sure this section reflects Tom's comments in the outline doc. 
https://docs.google.com/document/d/18XMcDQflwf7zdfjpjhMZYaBrvWXTsvaU0FuNZfU8yYo/edit#heading=h.wtgv99w5dsen

File: source/installation/kafka-deployments/migrator-with-kafka.txt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+Behavior
+--------
+
+Relational Migrator works with Kafka as a Kafka Connect sink connector. A 
+`Debezium connector <https://debezium.io/documentation/reference/stable/connectors/index.html>`__ 
+captures data events from your source database, while Relational Migrator 
+interprets and transforms events from the Debezium source and sends them to a 
+MongoDB cluster that acts as a data sink.
+
```
Comment (sarah-olson-mongodb) [inline source/installation/kafka-deployments/migrator-with-kafka.txt:32]: (graphic) 
Your outline states that you'll add a graphic here. I think that would be a really nice way to show the information.

I don't think this needs to be a blocking issue, however. If you'd like, please check with Tom and David to see if it makes sense to treat this as a fast follow update, once the bulk of your work has been completed and the docs are published.

File: source/installation/kafka-deployments/migrator-with-kafka.txt [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+You can use the following deployment methods to install Relational Migrator 
+with Kafka:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Deployment Method
+     - Description
+
+   * - :ref:`Install with Existing Kafka Cluster <install-kafka-cluster>`
```
Comment (sarah-olson-mongodb) [inline source/installation/kafka-deployments/migrator-with-kafka.txt:45]: (nit)
We should probably say Install with "an" Existing ...

**Review by sarah-olson-mongodb — COMMENTED** [review]
Nice work @jocelyn-mendez1. This is a great start! A few minor things for your consideration. Very happy to chat about any of this if you'd like!

File: source/installation/kafka-deployments/migrator-with-kafka.txt (hunk 3 above)
Comment (jocelyn-mendez1) [reply source/installation/kafka-deployments/migrator-with-kafka.txt]: Updated, thank you!

File: source/installation/kafka-deployments/install-confluent-kafka.txt [hunk 4]
```diff
@@ -0,0 +1,13 @@
+.. _install-confluent-kafka:
+
```
[the remark this answers is not in the document]
Comment (jocelyn-mendez1) [reply source/installation/kafka-deployments/install-confluent-kafka.txt:2]: Updated, thank you!

File: source/installation/kafka-deployments/migrator-with-kafka.txt (hunk 1 above)
Comment (jocelyn-mendez1) [reply source/installation/kafka-deployments/migrator-with-kafka.txt]: Updated.

File: source/installation/kafka-deployments/migrator-with-kafka.txt (hunk 2 above)
Comment (jocelyn-mendez1) [reply source/installation/kafka-deployments/migrator-with-kafka.txt:37]: I added a potential graphic!

**Comment by jocelyn-mendez1** [comment]
Thank you for your feedback @sarah-olson-mongodb! Let me know of any other possible updates.

**Review by sarah-olson-mongodb — APPROVED** [review]
Thanks so much for the update @jocelyn-mendez1. This looks really great. 
LGTM

Only minor nit is that you now use "improves resilience and scalability" twice, back to back. Let's pick just one location and go with that.

File: source/installation/kafka-deployments/migrator-with-kafka.txt [hunk 5]
```diff
@@ -0,0 +1,81 @@
+.. _kafka-intro:
+
+======================================
+Relational Migrator with Kafka Connect
```
Comment (tomhollander) [inline source/installation/kafka-deployments/migrator-with-kafka.txt:4]: Can we change this heading, maybe to something like "Integrating Relational Migrator with Kafka"? Kafka Connect is just one component of the solution.

File: source/installation/kafka-deployments/migrator-with-kafka.txt [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+   :class: singlecol
+
+You can use Relational Migrator to migrate data to and from Kafka. 
+Relational Migrator can be used as a Kafka Connect plug-in that improves 
+resilience and scalability of big data migration jobs.
+
+Use Cases
+---------
+
+Relational Migrator with Kafka is intended for large and long running jobs.
```
Comment (tomhollander) [inline source/installation/kafka-deployments/migrator-with-kafka.txt:26]: Can we provide a bit more detail here? Specifically, that Kafka provides built-in resilience, such that if any component of the solution is temporarily unavailable, the sync job will be able to recover and keep running.

File: source/installation/kafka-deployments/migrator-with-kafka.txt [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+   :name: genre
+   :values: reference
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+You can use Relational Migrator to migrate data to and from Kafka.
```
Comment (tomhollander) [inline source/installation/kafka-deployments/migrator-with-kafka.txt:19]: This sentence is inaccurate. Relational Migrator is still migrating data from a source relational database and writing to MongoDB. Kafka is basically being used as a robust transport layer between the components.

Also can we refer to the product as "Apache Kafka" on first mention?

**Review by tomhollander — CHANGES_REQUESTED** [review]
Great start, a few suggested updates.

File: source/installation/kafka-deployments/migrator-with-kafka.txt (hunk 5 above)
Comment (jocelyn-mendez1) [reply source/installation/kafka-deployments/migrator-with-kafka.txt]: Updated.

File: source/installation/kafka-deployments/migrator-with-kafka.txt (hunk 6 above)
Comment (jocelyn-mendez1) [reply source/installation/kafka-deployments/migrator-with-kafka.txt:28]: Updated.

File: source/installation/kafka-deployments/migrator-with-kafka.txt (hunk 7 above)
Comment (jocelyn-mendez1) [reply source/installation/kafka-deployments/migrator-with-kafka.txt]: Updated.

**Comment by davidsoulmdb** [comment]
Thanks to Tom's input, it's now +LGTM for me (is that like ATM machine? :)

**Comment by davidsoulmdb** [comment]
Two suggestions:

1. Suggest breaking the tech part out and expanding slightly?
"Relational Migrator works with Kafka as a Kafka Connect sink connector. A 
[Debezium connector](https://debezium.io/documentation/reference/stable/connectors/index.html)
 captures data events from your source database, while Relational Migrator interprets and transforms events from the Debezium source and sends them to a MongoDB cluster that acts as a data sink."
->
"Architecturally, Relational Migrator is deployed within Kafka as a Kafka Connect sink connector. Database events are captured using the open-source [Debezium tool](https://debezium.io/documentation/reference/stable/connectors/index.html) for database change data capture. A Debezium Connector captures these events from your source database, 

3. Tasks -> Deployment Options

**Comment by davidsoulmdb** [comment]
Otherwise LTGM

**Review by tomhollander — APPROVED** [review]

**Pull request merged**